conda/common/path.py

```diff
@@ -85,14 +83,13 @@ def tokenized_startswith(test_iterable, startswith_iterable):
     return all(t == sw for t, sw in zip(test_iterable, startswith_iterable))
 
 
-def get_all_directories(files):
+def get_all_directories(files: Iterable[str]) -> list[tuple[str]]:
     return sorted(set(tuple(f.split('/')[:-1]) for f in files) - {()})
 
 
-def get_leaf_directories(files):
-    # type: (List[str]) -> List[str]
-    # give this function a list of files, and it will hand back a list of leaf directories to
-    #   pass to os.makedirs()
+def get_leaf_directories(files: Iterable[str]) -> Sequence[str]:
```

Comment:
Let's use os.PathLike when we are expecting paths:

```suggestion
def get_leaf_directories(files: Iterable[os.PathLike]) -> tuple[str, ...]:
```